### PR TITLE
BaSP-MR-83: super admin implement shared components

### PR DESCRIPTION
**Pull request by ghost**
super admin implement shared components

Brief Descripition
Integrate the shared components into Super-Admin component, replacing any redundant components implementations.

The last commit was done by Emanuel as we did it from his PC in the co-working room.

### CODE REVIEW

File: src/Components/SuperAdmins/Form/index.jsx [hunk 1]
```diff
@@ -61,22 +70,50 @@ const Form = () => {
...
+  //       setTitleModal('Error');
+  //       setBodyModal(data.message);
+  //       setShowModal(true);
+  //     }
+  //   } catch (error) {
+  //     console.error(error);
+  //     throw new Error('An error has occurred while deleting the Super Admin');
+  //   }
+  // };
+
```
Comment (EmaBaldor) [inline src/Components/SuperAdmins/Form/index.jsx]: This shouldn't be here

**Comment by lorenzomauro59** [comment]
[[Super-Admin] - Implement the shared components](https://trello.com/c/l6AucEQE/83-super-admin-implement-the-shared-components)

**Comment by marcohzo** [comment]
When creating a new SuperAdmin and encountering an error, closing the modal after opening it redirects me back to the SuperAdmin page, which also closes the form. To provide a more consistent experience and allow users to easily make necessary corrections, it would be great if we could keep the form open even after closing the modal.

**Review by marcohzo — COMMENTED** [review]
When creating a new SuperAdmin and encountering an error, closing the modal after opening it redirects me back to the SuperAdmin page, which also closes the form. To provide a more consistent experience and allow users to easily make necessary corrections, it would be great if we could keep the form open even after closing the modal.

**Review by marcohzo — APPROVED** [review]

**Review by Federico-Peralta — COMMENTED** [review]
When i make an error on the edit or add and click the button of the modal it redirects me to the table

File: src/Components/SuperAdmins/Form/index.jsx [hunk 2]
```diff
@@ -98,60 +116,88 @@ const Form = () => {
     history.push('/super-admins');
   };
 
+  const closeModal = () => {
+    setShowModal(false);
+    setShowSuccessModal(false);
+    history.replace('/super-admins');
```
Comment (fernandofernandezuy) [inline src/Components/SuperAdmins/Form/index.jsx]: This replace shouldn't be on the closeModal, put some logic into the functions addSuperAdmin() and updateSuperAdmin() when the response is okay, in this way when i make a mistake doing the form doesn't redirect me back to the table.

File: src/Components/SuperAdmins/index.jsx [hunk 3]
```diff
@@ -59,24 +78,44 @@ const SuperAdmins = () => {
...
-        </button>
+        <h2>Super Admins</h2>
+        <Button text="Add New" clickAction={handleAddSuperAdmin} type="add" />
       </div>
       <Table
-        data={superAdmins}
-        handleDeleteItem={handleDeleteSuperAdmin}
+        data={superAdmins || []}
+        properties={['firstName', 'email', 'password']}
+        columnTitles={['First Name', 'Email', 'Password']}
```
Comment (fernandofernandezuy) [inline src/Components/SuperAdmins/index.jsx]: We shouldn't display the password in the table.

**Review by fernandofernandezuy — CHANGES_REQUESTED** [review]

**Comment by ale-schmidt** [comment]
I was reading my mates comments and I agree with all of them :)

File: src/Components/SuperAdmins/Form/index.jsx [hunk 4]
```diff
@@ -61,10 +72,17 @@ const Form = () => {
...
       } else {
-        alert(data.message);
+        setTypeStyle('error');
+        setTitleModal('Error');
+        setBodyModal(data.message);
+        setShowModal(true);
+        setShowSuccessModal(false);
       }
     } catch (error) {
       console.error(error);
```
Comment (ale-schmidt) [inline src/Components/SuperAdmins/Form/index.jsx:93]: I would recommend not to leave any console.log in the final code

Comment (ghost) [reply src/Components/SuperAdmins/Form/index.jsx:93]: no is a console.log.. is .error!

**Review by ale-schmidt — APPROVED** [review]
I was checkin out the code and all looks working as expected :)

**Review by lorenzomauro59 — APPROVED** [review]
Only the comment that Fer made! Rest is Ok!!

**Review by fernandofernandezuy — APPROVED** [review]

**Comment by nazarenapantanetti** [comment]
Tested and approved! All the shared components considered in Trello's ticket were included.

**Pull request merged**